modules/telemetry/telemetry/src/metric_record.cpp

```
@@ -126,7 +127,11 @@ MetricRecorder::MetricRecorder()
 }
 
 MetricRecorder::~MetricRecorder() {
-  entries_consumer_.stop();
+  try {
+    entries_consumer_.stop().wait();
```

```suggestion
    entries_consumer_.stop().get();
```

---

whats the difference in this case? return value is void right?

---

you need to call `get` on a future to surface potential exception from inside the asynch

---

crazy so exceptions are suppressed when calling wait()?

---

not really, the point of `wait` is that eventually you call `get`. You can call `wait` many time on the same object, but once you call `get` the future is no longer valid.